### PR TITLE
add concept of blocking jobs

### PR DESCRIPTION
**Pull request by sairez**
for one-run-only jobs during deployment

### CODE REVIEW

File: k8s-scripts.config [hunk 1]
```diff
@@ -24,3 +24,5 @@ DEPLOYMENTS=('example-app')
 # List of files ending in '.job.yml' in the kube directory (Not supported yet)
 JOBS=()
 
+# List of files ending in '.blockjob.yml' in the kube directory (Not supported yet)
```
Comment (philipcristiano) [inline k8s-scripts.config]: This looks like it can be updated as this PR will add the functionality right?

Comment (philipcristiano) [inline k8s-scripts.config]: +1 on updating docs as well!

File: bin/k8s-deploy [hunk 2]
```diff
@@ -56,6 +56,51 @@ do
...
+done
+echo "Done deploying Blocking Jobs"
+echo ""
+
+echo "Running Blocking Jobs"
+for index in "${!BLOCKING_JOBS[@]}"
+do
+  BLOCKING_JOB=${BLOCKING_JOBS[$index]##*/}
+  echo "Checking jobs for $BLOCKING_JOB"
+  timeout.sh -t ${DEPLOY_TIMEOUT} run-blocking-job $BLOCKING_JOB $NAMESPACE
```
Comment (philipcristiano) [inline bin/k8s-deploy]: Should we use a different timeout for this knowing that we will likely be running migrations or similar here that may take a different amount of time then we would expect for a deployment? By default could be set to the same value.

File: README.md [hunk 3]
```diff
@@ -42,6 +42,9 @@ DEPLOYMENTS=('example-app')
 
 # List of files ending in '.job.yml' in the kube directory (Not supported yet)
 JOBS=()
+
+# List of files ending in '.blockingjob.yml' in the kube directory (Not supported yet)
```
Comment (philipcristiano) [inline README.md]: This are supported with this PR right?

File: bin/k8s-deploy (hunk 2 above)
Comment (sairez) [inline bin/k8s-deploy]: Hmm, fair point. It does seem reasonable to have a separate timeout. A shared, sane default would be good.

**Pull request merged**